cmake/Modules/FindDPDK.cmake

```diff
@@ -0,0 +1,181 @@
+# ~~~
+# - Try to find DPDK include dirs and libraries
+#
+# Find the DPDK includes and client library
+# This module defines:
+#  DPDK_FOUND, If false, do not try to use DPDK.
+#  DPDK_INCLUDE_DIRS, where to find rte_config.h and rte_version.h
+#  DPDK_LIBRARIES, the libraries needed by a DPDK user
+#  DPDK_CFLAGS_OTHER, the compile flags to use
+#  DPDK_VERSION, the version of the library
+# ~~~
+
+include(FindPackageHandleStandardArgs)
+
+function(DPDK_READ_VERSION DPDK_VERSION DPDK_VERSION_FILE)
+  if(NOT DPDK_VERSION_FILE)
+    return()
+  endif()
+
+  file(READ "${DPDK_VERSION_FILE}" DPDK_VERSION_STR)
+  string(
+    REGEX MATCH
+          "#define RTE_VER_YEAR ([0-9]+)"
+          _
+          ${DPDK_VERSION_STR})
+  set(DPDK_VERSION_MAJOR ${CMAKE_MATCH_1})
+
+  string(
+    REGEX MATCH
+          "#define RTE_VER_MONTH ([0-9]+)"
+          _
+          ${DPDK_VERSION_STR})
+  set(DPDK_VERSION_MINOR ${CMAKE_MATCH_1})
+
+  string(
+    REGEX MATCH
+          "#define RTE_VER_MINOR ([0-9]+)"
+          _
+          ${DPDK_VERSION_STR})
+  set(DPDK_VERSION_PATCH ${CMAKE_MATCH_1})
+
+  set(DPDK_VERSION
+      "${DPDK_VERSION_MAJOR}.${DPDK_VERSION_MINOR}.${DPDK_VERSION_PATCH}"
+      PARENT_SCOPE)
+endfunction()
+
+# Try to find DPDK with pkg-config first!
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(
+    DPDK
+    QUIET
+    IMPORTED_TARGET
+    libdpdk>=18.11)
+endif()
+
+# We found using Pkg-Config!
+if(DPDK_FOUND)
+  message("DPDK found with pkg-config")
+  if(DPDK_DEBUG)
+    message("-----------")
+    message("Libraries: ${DPDK_LIBRARIES}")
+    message("Link Libraries: ${DPDK_LINK_LIBRARIES}")
+    message("Library DIr: ${DPDK_LIBRARY_DIRS}")
+    message("Ldflags: ${DPDK_LDFLAGS}")
+    message("Include Dirs: ${DPDK_INCLUDE_DIRS}")
+    message("Cflags: ${DPDK_CFLAGS}")
+    message("Cflags Other: ${DPDK_CFLAGS_OTHER}")
+    message("Version: ${DPDK_VERSION}")
+    message("-----------")
+  endif()
+else()
+  message("DPDK not found with pkg-config trying legacy mode")
+
+  # Find the include dirs and get the version
+  find_path(DPDK_VERSION_INCLUDE_DIR rte_version.h REQUIRED PATH_SUFFIXES dpdk include)
+  find_path(DPDK_CONFIG_INCLUDE_DIR rte_config.h REQUIRED PATH_SUFFIXES dpdk include)
+
+  set(DPDK_INCLUDE_DIRS ${DPDK_INCLUDE_DIRS})
+  list(APPEND DPDK_INCLUDE_DIRS ${DPDK_VERSION_INCLUDE_DIR})
+  list(APPEND DPDK_INCLUDE_DIRS ${DPDK_CONFIG_INCLUDE_DIR})
+  list(REMOVE_DUPLICATES DPDK_INCLUDE_DIRS)
+
+  dpdk_read_version(DPDK_VERSION "${DPDK_VERSION_INCLUDE_DIR}/rte_version.h")
+
+  # Get all the libraries regarding the version
+  list(
+    APPEND
+    _DPDK_LOOK_FOR_LIBS
+    net
+    kni
+    ethdev
+    mbuf
+    eal
+    mempool
+    ring
+    kvargs
+    hash
+    cmdline
+    pci
+    bus_pci
+    bus_vdev
+    mempool_ring)
+
+  if(DPDK_VERSION VERSION_LESS "20.11")
+    list(
+      APPEND
+      _DPDK_LOOK_FOR_LIBS
+      pmd_bond
+      pmd_vmxnet3_uio
```

Comment:
I tried to compile it with DPDK 18.11.11 without `pkgconfig` and I got this linker error:
```shell
[ 51%] Built target Pcap++
Scanning dependencies of target Arping
[ 51%] Building CXX object Examples/Arping/CMakeFiles/Arping.dir/main.cpp.o
[ 52%] Linking CXX executable ../../examples_bin/Arping
/usr/bin/ld: cannot find -lrte_pmd_vmxnet3_uio-NOTFOUND
collect2: error: ld returned 1 exit status
make[2]: *** [Examples/Arping/CMakeFiles/Arping.dir/build.make:112: examples_bin/Arping] Error 1
make[1]: *** [CMakeFiles/Makefile2:1731: Examples/Arping/CMakeFiles/Arping.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

It can't find `-lrte_pmd_vmxnet3_uio`, it only has `librte_pmd_vmxnet3`

```shell
$ ll | grep vmxnet
-rw-r--r--  1 root root   68690 Dec  9 07:13 librte_pmd_vmxnet3.a
lrwxrwxrwx  1 root root      37 Dec  9 08:08 librte_pmd_vmxnet3.so -> dpdk/pmds-18.11/librte_pmd_vmxnet3.so*
lrwxrwxrwx  1 root root      39 Dec  9 08:08 librte_pmd_vmxnet3.so.1 -> dpdk/pmds-18.11/librte_pmd_vmxnet3.so.1*
lrwxrwxrwx  1 root root      41 Dec  9 08:08 librte_pmd_vmxnet3.so.1.1 -> dpdk/pmds-18.11/librte_pmd_vmxnet3.so.1.1*
```

Reply:
this seems to make the CI failed with the actual dockers https://github.com/seladb/PcapPlusPlus/actions/runs/3660688226/jobs/6188110607

Reply:
Do we really need this library ?

Reply:
I did some research. There are 2 ways to compile DPDK 18.11 and 19.11: 
1. With `make` as we currently do: https://github.com/seladb/PcapPlusPlus-DockerImages/blob/f6ca806360739f0d4c79fda8ffbd216edb62bcd2/Dockerfile-ubuntu2004-dpdk1811#L37
2. With `meson` and `ninja` as we do in newer DPDK versions: https://github.com/seladb/PcapPlusPlus-DockerImages/blob/f6ca806360739f0d4c79fda8ffbd216edb62bcd2/Dockerfile-ubuntu2004-dpdk2011#L32

Apparently there is a difference between the 2 build systems: the first generates `librte_pmd_vmxnet3_uio` and the second generates `librte_pmd_vmxnet3`. Since these are older versions I think we can keep it as is and assume people build DPDK using `make`.

Reply:
So... nothing to fix here 😃

Reply:
This the downside of the specific loop pattern: you cannot deal easily with variants. However, there options:

Reply:
this should be good with latest version